### ISSUE TITLE
[R] Unable to use write_parquet within Docker container running R

### ISSUE THREAD

**Issue by Gustavo-CR** [apache/arrow#14707]
Hello all, 

I have a docker image running R version 3.6.3: [gcera/r_ver:3.6.3](https://hub.docker.com/layers/gcera/r_ver/3.6.3/images/sha256-b0f6453bdd77e1a2d4f3b6092107057a7aad0f8e4c8dd148b646ef8d65722cc4?context=repo). 
I am able to install arrow 0.17.0, but not able to use `write_parquet` function with the install_arrow() message popping up. I have gone through the "[Installing the Arrow Package on Linux](https://arrow.apache.org/docs/r/articles/install.html)" guide with little luck after changing the install configuration options. Basically, I get the following message every time: 
```
Error in Table__from_dots(dots, schema) :
  Cannot call Table__from_dots(). Please use arrow::install_arrow() to install required runtime libraries.
```

I have read on previous issues that you can integrate Docker images into CI setup. Could you tell me a bit more about this? I have included my image at the beginning, so I hope that helps. 

Here is the R session info: 
```
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 10 (buster)

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.3.5.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] arrow_0.17.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0 bit_1.1-15.2     compiler_3.6.3   magrittr_2.0.1
 [5] assertthat_0.2.1 R6_2.5.0         glue_1.4.2       Rcpp_1.0.3
 [9] bit64_0.9-7      vctrs_0.3.7      rlang_0.4.10     purrr_0.3.3
```

### Component

R

**Comment by assignUser** [comment]
Is there a reason why you are installing a super ancient version of arrow? (0.17.0 >2 years old, we are at 10.0.0 now).

I had a look at the container, it has an mran snapshot of cran hard coded:
```
> install.packages("arrow")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
trying URL 'https://mran.microsoft.com/snapshot/2020-04-24/src/contrib/arrow_0.17.0.tar.gz'
```

This is what prevents you from installing a recent version of arrow. If there is no specific reason for this I would recommend changing this to a recent snapshot or RSPM or vanillar CRAN. It is possible to force installation of arrow from CRAN with
`Sys.setenv(LIBARROW_BINARY = TRUE); install.packages("arrow", repos=c(CRAN="https://cran.rstudio.com"))` but you will also need to install arrow's dependencies with this method to make sure they are of a recent, compatible version.

> I have read on previous issues that you can integrate Docker images into CI setup. 

That is possible but the arrow project is not really the right avenue for this general question.  It might also not be necessary unless you have specialized use case (e.g. specific hard to install system dependencies). I recommend checking out the suite of R GitHub actions in https://github.com/r-lib/actions

**Comment by Gustavo-CR** [comment]
Hi, thanks a lot for your answer. 

It looks like the MRAN snapshot remained as default after I was installing a few other packages with older versions. 
After pointing to the CRAN repo to install the latest arrow version, I discover the arrow package is not available for my docker R version - 3.6.3. I will try and update to a more recent R version to see if I can install arrow. 

Thanks for pointing out the repo conflict!
